deployment/docker-compose.yml

```diff
@@ -0,0 +1,124 @@
+    expose:
+     - 9000
+    # ports:
+    #   - 9000:9000
+    depends_on:
+      - postgresvectordb
+    volumes:
+     - logs-vol:/app/logs
+     - chroma-db:/app/app/chromadb_store
+    networks:
+     - chatbot-network
+
+  postgresvectordb:
+    image: ankane/pgvector
+    ports:
+      - "5435:5432"
+    environment:
+      - POSTGRES_PASSWORD=${POSTGRES_DB_PASSWORD:-secret}
+      - POSTGRES_USER=${POSTGRES_DB_USER:-admin}
+      - POSTGRES_DB=${POSTGRES_DB_NAME:-adotbcollection}
+      - POSTGRES_HOST_AUTH_METHOD=md5
+    restart: always
+    expose:
+      - 5432
+    networks:
+     - chatbot-network
+    volumes:
+     - postgres-db-vol:/var/lib/postgresql/data
+     - postgres-db-backup:/var/backups:rw
+    labels:
+      ofelia.enabled: "true"
+      ofelia.job-exec.backup.schedule: "0 0 0 * * *" #"Takes a single dump within the db container every day at 00:00 am"
+      ofelia.job-exec.backup.command: "pg_dump --dbname=postgresql://$POSTGRES_USER:$POSTGRES_PASSWORD@localhost:5432/$POSTGRES_DB  --file=/var/backups/postgres_backup_latest.sql"
+
+  webserver:
+    image: nginx:latest
+    ports:
+     - 80:80
+     - 443:443
+    restart: always
+    expose:
+     - 80
+     - 443
+    depends_on:
+     - chatbot
+    environment:
+      - CHAT_DOMAIN=${DOMAIN:-"localhost"}
+    volumes:
+     - ./nginx/nginx.conf.template:/etc/nginx/templates/default.conf.template:ro
+     # - ./nginx/nginx.conf:/etc/nginx/nginx.conf:ro
+     -  logs-vol:/var/log/nginx/
+     - ./certbot/www:/var/www/certbot/:ro
+     - ./certbot/conf/:/etc/nginx/ssl/:ro
+    networks:
+     - chatbot-network
+
+  certbot:
+    image: certbot/certbot:latest
+    volumes:
+     - ./certbot/www/:/var/www/certbot/:rw
+     - ./certbot/conf/:/etc/letsencrypt/:rw
+    networks:
+     - chatbot-network
+
+  ofelia-scheduler:
+    image: mcuadros/ofelia:v0.3.7
+    depends_on:
+      - postgresvectordb
+    command: daemon --docker
+    volumes:
+      - /var/run/docker.sock:/var/run/docker.sock:ro
+      - postgres-db-backup:/var/backups:r
+      - ./backup:/var/backup-copies/:rw
+    labels:
+      ofelia.job-local.copy-backup-daily.schedule: "0 30 0 * * *" # Copies latest dump from DB contaier every day at 00:30 am
+      ofelia.job-local.copy-backup-daily.command: sh -c "cp /var/backups/postgres_backup_latest.sql /var/backup-copies/postgres_backup_daily_`TZ=\":Asia/Calcutta\" date +%Y-%m-%d_%H_%M_%S`.sql"
+      ofelia.job-local.prune-daily-backup.schedule: "0 45 0 * * *" # Retains only latest 7 of the daily backups 
+      ofelia.job-local.prune-daily-backup.command: sh -c "ls -tr /var/backup-copies/postgres_backup_daily* | head -n -7 |  xargs rm -f --"
+      ofelia.job-local.copy-backup-weekly.schedule: "0 30 0 ? * SUN" # Copies latest dump on every sunday 00:30 am
+      ofelia.job-local.copy-backup-weekly.command: sh -c "cp /var/backups/postgres_backup_latest.sql /var/backup-copies/postgres_backup_weekly_` TZ=\":Asia/Calcutta\" date +%Y-%m-%d_%H_%M_%S`.sql"
```

Comment:
It might be worth adding some logging in case errors show up. I asked ChatGPT what it looks like to add error logging for Ofelia jobs and it gave these examples:

For the copy jobs:
```
ofelia.job-local.copy-backup-daily.command: sh -c "cp /var/backups/postgres_backup_latest.sql /var/backup-copies/postgres_backup_daily_`TZ=\":Asia/Calcutta\" date +%Y-%m-%d_%H_%M_%S`.sql && echo 'Copy successful' >> /var/log/backup.log || echo 'Copy failed' >> /var/log/backup.log"
```

For the prune jobs:
```
ofelia.job-local.prune-daily-backup.command: sh -c "ls -tr /var/backup-copies/postgres_backup_daily* | head -n -7 | xargs rm -f -- && echo 'Prune successful' >> /var/log/backup.log || echo 'Prune failed' >> /var/log/backup.log"
```

We might need/want a specific command for the log files.

This is just an idea. I'll defer to your judgement and expertise!

Reply:
ChatGPT's suggestions were very helpful :smile: